### PR TITLE
[libc++] Remove __unconstrained_reverse_iterator

### PR DESCRIPTION
**Pull request by philnik777**
`__unconstrained_reverse_iterator` has outlived its usefullness, since the standard and subsequently the compilers have been fixed.

### CODE REVIEW

**Review by ldionne — APPROVED** [review]
LGTM assuming we only support compilers where this has been fixed. Don't we have to wait a bit more before we merge this? We still support Clang 16 (not for long though).

**Comment by philnik777** [comment]
> LGTM assuming we only support compilers where this has been fixed. Don't we have to wait a bit more before we merge this? We still support Clang 16 (not for long though).

The fix has been implemented in Clang 16, so we could have removed it a while ago already.

**Pull request merged**
